cmd/arrai/build.go

```diff
@@ -0,0 +1,151 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/arr-ai/arrai/pkg/arrai"
+	"github.com/arr-ai/arrai/pkg/arraictx"
+	"github.com/arr-ai/arrai/pkg/ctxfs"
+	"github.com/arr-ai/arrai/rel"
+	"github.com/arr-ai/arrai/syntax"
+	"github.com/arr-ai/arrai/tools"
+	"github.com/arr-ai/wbnf/parser"
+	"github.com/spf13/afero"
+	"github.com/urfave/cli/v2"
+)
+
+var compileCommand = &cli.Command{
+	Name:    "compile",
+	Aliases: []string{"compile"},
+	Usage:   "compile arrai scripts into a runnable binary",
+	Action:  compile,
+	Flags: []cli.Flag{
+		outFlag,
+	},
+}
+
+func compile(c *cli.Context) error {
+	tools.SetArgs(c)
+	file := c.Args().Get(0)
+
+	return compileFile(arraictx.InitRunCtx(context.Background()), file, c.Value("out").(string))
+}
+
+func compileFile(ctx context.Context, path, out string) error {
+	if err := runFileExists(ctx, path); err != nil {
+		return err
+	}
+
+	bundledScripts := bytes.Buffer{}
+	if err := bundleFiles(ctx, path, &bundledScripts, ""); err != nil {
```

Comment:
Nit: passing an empty string as an arg is a bit of a smell. Maybe refactor `bundleFiles` so that `bundleFiles` handles this simple case, and there's some other `bundleFilesTo` that takes a non-empty `out` param? Something like that.